### ISSUE TITLE
Saving TICK scripts in Chronograf makes human task name disapear

### ISSUE THREAD

**Issue by stefanhorning** [influxdata/chronograf#4304]
I am using Chronograf 1.6.1 and when I edit existing tick scripts (initially created by the task GUI of Chronograf) the human friendly name is lost and instead in the list view only the internal TICK id is visible with this kind of format: `chronograf-v1-30ac70ed-e070-4919-b21b-e375ebf81a1b`.

Steps to reproduce:
1. Click "Build alert rule" in the "Manage Task" view and give the rule a nice name and save it
2. Back on the "Manage Tasks" page scroll down to the "TICKscripts" list
3. Open that same rule in the TICK editor, make a change to it and save again.
4. Back in the list view the rule has a) dissapeared from the "Alert Rules" list and b) lost the human readable name in the "TICKscript" list.

If I remember correctly this wasn't always like this, as I prevously was able to make small adjustments (like changing the query conditions) to rules using the TICK editor without changing the listing or the names.

**Comment by stefanhorning** [comment]
Apparently that doesn't happen on any change, could reproduce it though when changing tagging conditions in a query.

**Comment by stefanhorning** [comment]
Can only reproduce the error when I change this line

```
var whereFilter = lambda: ("some_tag" == 'foo')
```
to 

```
var whereFilter = lambda: ("some_tag" == 'foo') AND ("server" != 'https://some-domain.com/health_check')
```

Working with tags here to exclude on server from http_response checks in the alert.

**Comment by russorat** [comment]
@stefanhorning behind the scenes, we have to store the human name outside of kapa due to some limitations in the api. we will investigate this.

**Comment by jregovic** [comment]
Any updates on this? 

We still see this in 1.7.7

**Comment by fomenta** [comment]
Any updates on this?

**Comment by jpm-izinto** [comment]
I'm also experiencing this. It makes managing tasks in Chronograf very difficult as I don't know from the name what the script does.

**Comment by HeikoOnnebrink** [comment]
Same here with latest version .. once we add a string function in tick editor name swaps to v1-uuid style.. having no human readable script names is a no-go

**Comment by russorat** [comment]
The UI of chronograf pulls the name for the UI from the `var name` variable in the TICKscript, and defaults to the uuid if it is not present. 

![Kapture 2020-09-24 at 11 45 43](https://user-images.githubusercontent.com/4805997/94186529-a9a71580-fe5b-11ea-966e-3e5e7c3af9e4.gif)

**Comment by HeikoOnnebrink** [comment]
@russorat  Thx for reopening this issue 👍 

We always set the var name in our tick scripts, in this example we set the name '**WAL SHIPPING DELAY**'

<img width="824" alt="image" src="https://user-images.githubusercontent.com/19688787/94226014-ccd2d300-fef6-11ea-8b0e-eed165c464b8.png">

But script shows up in v1-UUID style and is not present in alert rules list


<img width="1059" alt="image" src="https://user-images.githubusercontent.com/19688787/94226074-f0961900-fef6-11ea-8509-58b835c41d50.png">

Once I remove the substring function from where condition it saves it with the name as defined in var name :

**change:**
`var whereFilter = lambda: (strContains("directory", 'pg_wal') OR strContains("directory", 'pg_xlog'))`
**to:**
`var whereFilter = lambda: ("directory" == '/some/folder/pg_wal' OR "directory" == 'some/folder/pg_xlog')`

<img width="1236" alt="image" src="https://user-images.githubusercontent.com/19688787/94226384-c3963600-fef7-11ea-9970-a9507f451a1c.png">

**Comment by HeikoOnnebrink** [comment]
Here a copy of the full tick script 

`var db = 'postgresql'

var rp = 'default'

var measurement = 'filecount'

var groupBy = ['ID', 'directory']

var whereFilter = lambda: (strContains("directory", 'pg_wal') OR strContains("directory", 'pg_xlog'))

var name = 'WAL SHIPPING DELAY'

var idVar = name + '-{{.Group}}'

var message = '$db-uuid$ {{ index .Tags "ID" }}
$message$ THE WAL SHIPPING DELAY IS MORE THAN {{ index .Fields "value"}}B
'

var messageok = '$db-uuid$ {{ index .Tags "ID" }}
$message$ THE WAL SHIPPING DELAY WENT BACK TO NORMAL
'

var idTag = 'alertID'

var levelTag = 'level'

var messageField = 'message'

var durationField = 'duration'

var outputDB = 'chronograf'

var outputRP = 'autogen'

var outputMeasurement = 'alerts'

var triggerType = 'threshold'

var warn = 4000000000

var crit = 6000000000

var data = stream
    |from()
        .database(db)
        .retentionPolicy(rp)
        .measurement(measurement)
        .groupBy(groupBy)
        .where(whereFilter)
    |eval(lambda: "size_bytes")
        .as('value')

var trigger = data
    |alert()
        .warn(lambda: "value" > warn)
        .crit(lambda: "value" > crit)
        .message('{{ if eq .Level "OK" }} ' + messageok + ' {{ else }} ' + message + ' {{ end }}')
        .id(idVar)
        .idTag(idTag)
        .levelTag(levelTag)
        .messageField(messageField)
        .durationField(durationField)
        .stateChangesOnly()
        .post()
        .endpoint('rdbalert')

trigger
    |eval(lambda: "value")
        .as('value')
        .keep()
    |influxDBOut()
        .create()
        .database(outputDB)
        .retentionPolicy(outputRP)
        .measurement(outputMeasurement)
        .tag('alertName', name)
        .tag('triggerType', triggerType)

trigger
    |httpOut('output')
`

**Comment by HeikoOnnebrink** [comment]
and here the exact version we are using for all components (all deployed as docker container)

kapacitor:1.5.6
chronograf:1.8.2
influxdb:1.8.2

**Comment by afausti** [comment]
@russorat we are seeing this issue here too (Chronograf 1.8.4).

I cannot reproduce it consistently but I noticed that if you create a new TICKscript using the "+ Write TICKscript" button (without going through the "Alert Rule builder") the name that is displayed is the name that you typed in the text box - which is referred as `Task ID` (it does not use the `name` variable).

And `Task ID` must follow these rules:

```
Task ID must contain only letters, numbers, '-', '.' and '_'. "THIS IS THE NAME I TYPED" 
```

However, if you use the Alert rule builder, it sets the variable `name` and uses that as display name. At this point I think a task ID like `chronograf-v1-30ac70ed-e070-4919-b21b-e375ebf81a1b` is automatically generated. That internal name happens to be used as display name sometimes (e.g. after you edit the script and save it) but not always.

Hope this provides more evidence to where the problem is...

**Comment by DarthSidious-DBA** [comment]
@russorat I am also facing the same problem (1.8.2).

It is impossible to verify the different alerts (more than 20) when having this kind of tick names.
That´s frustrating.
I guess it changed from readable to other notation when using a string funtion in WHERE-Clause.

Hope it will be fixed soon.
Thanks in advance.

**Comment by russorat** [comment]
we have this scheduled for the next maintenance release which should be out by the end of Nov, maybe sooner.

**Comment by stefanhorning** [comment]
After not running into this for quite some time, I can now reproduce this again.

**Comment by sranka** [comment]
@stefanhorning ... this is fixed since 1.8.8

**Comment by a-vogel-tappert** [comment]
Now works fine for me with version 1.8.8 of Chronograf
